src/main/java/net/snowflake/ingest/utils/HttpUtil.java

```diff
@@ -294,7 +294,8 @@ static HttpRequestRetryHandler getHttpRequestRetryHandler() {
       if (exception instanceof NoHttpResponseException
           || exception instanceof javax.net.ssl.SSLException
           || exception instanceof java.net.SocketException
-          || exception instanceof java.net.UnknownHostException) {
+          || exception instanceof java.net.UnknownHostException
+          || exception instanceof java.net.SocketTimeoutException) {
```

Comment:
Do we have a good reason to add this? Just want to make sure we are not blindly adding this that masks any bugs in the code.

Reply:
Yeah, this is a network connectivity issue, which we are not retrying, which leads to channel invalidation.